### PR TITLE
LVGL2-review

### PR DESCRIPTION
**Pull request by marqdevx**
Added USB handler for the Portenta M4

For tutorials:
- [ ] I double checked the wiring in the schematics/images
- [ ] I exported the cover image without background
- [ ] I used SVG format for pure vector images
- [ ] I updated the metadata.json file of the tutorial
- [ ] I updated the metadata.json file of the previous tutorial to contain the "next" link
- [ ] I did a spell checking before requesting a review
- [ ] I used title case for titles

### CODE REVIEW

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md [hunk 1]
```diff
@@ -354,7 +354,79 @@ To finish with the functions, the mouse pointer for the LVGL mouse driver, save
 
 Once we have added these functions, our interface will be created. Then, we need to go back to the main tab of the `lvgl_interaction.ino` sketch to finish the programming of the interface.
 
-## 4. Add the Mouse and Keyboard receivers
+## 4. Create the Mouse and Keyboard RPC senders
```
Comment (Lenardgeorge) [inline content/tutorials/portenta-h7/por-ard-lvgl2/content.md:357]: Should be H3 title

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md [hunk 2]
```diff
@@ -354,7 +354,79 @@ To finish with the functions, the mouse pointer for the LVGL mouse driver, save
...
+* Mouse
+    * `mouseCallback(tusbh_ep_info_t* ep, const uint8_t* mouse)`  
+        This function handle the mouse buttons and the relative change of position
+* Keyboard
+    * `keyboardCallback(tusbh_ep_info_t* ep, const uint8_t* keys)`  
+        Get the input keys, proccess them and send the correct character.
+
+**Note**: This functions are defined in the `Portenta_USBhost.h` wrapper, because it needs to be defined for the USB API structure of the USB library, defining them its needed to create the logic of those events.
+
+### Include needed libraries
```
Comment (Lenardgeorge) [inline content/tutorials/portenta-h7/por-ard-lvgl2/content.md:373]: Remove the subheading from here. We don't have subheading inside Steps

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md [hunk 3]
```diff
@@ -354,7 +354,79 @@ To finish with the functions, the mouse pointer for the LVGL mouse driver, save
...
+
+The sketch will contain the next functions
+* Mouse
+    * `mouseCallback(tusbh_ep_info_t* ep, const uint8_t* mouse)`  
+        This function handle the mouse buttons and the relative change of position
+* Keyboard
+    * `keyboardCallback(tusbh_ep_info_t* ep, const uint8_t* keys)`  
+        Get the input keys, proccess them and send the correct character.
+
+**Note**: This functions are defined in the `Portenta_USBhost.h` wrapper, because it needs to be defined for the USB API structure of the USB library, defining them its needed to create the logic of those events.
```
Comment (Lenardgeorge) [inline content/tutorials/portenta-h7/por-ard-lvgl2/content.md:371]: Do we have a reference page to the `Portenta_USBhost.h` wrapper ?

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md [hunk 4]
```diff
@@ -354,7 +354,79 @@ To finish with the functions, the mouse pointer for the LVGL mouse driver, save
...
+```cpp
+    #include "Portenta_USBhost.h"
+    #include "RPC_internal.h"
+
+    #ifndef CORE_CM4
+    #error "This sketch should be compiled for Portenta (M4 core)"
+    #endif
+```
+
+### Initialization
```
Comment (Lenardgeorge) [inline content/tutorials/portenta-h7/por-ard-lvgl2/content.md:387]: Remove subheading

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md [hunk 5]
```diff
@@ -354,7 +354,79 @@ To finish with the functions, the mouse pointer for the LVGL mouse driver, save
...
+
+The setup will initialize the USB protocol and look for the connected USB hub
+
+```cpp
+    void setup() {
+        // put your setup code here, to run once:
+        USBhost_init();
+    }
+```
+### Mouse handler
```
Comment (Lenardgeorge) [inline content/tutorials/portenta-h7/por-ard-lvgl2/content.md:397]: Remove subheading

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md [hunk 6]
```diff
@@ -354,7 +354,79 @@ To finish with the functions, the mouse pointer for the LVGL mouse driver, save
...
+```cpp
+    static int mouseCallback(tusbh_ep_info_t* ep, const uint8_t* mouse) {
+        uint8_t btn = mouse[0];
+        int8_t x = ((int8_t*)mouse)[1];
+        int8_t y = ((int8_t*)mouse)[2];
+        RPC1.call("on_mouse", btn, x, y);
+    }
+```
+
+### Keyboard Handler
```
Comment (Lenardgeorge) [inline content/tutorials/portenta-h7/por-ard-lvgl2/content.md:415]: Remove the subheading

**Review by Lenardgeorge — COMMENTED** [review]
Need to correct the title formats and remove the subheadings from each of the instruction steps.

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md (hunk 1 above)
Comment (marqdevx) [inline content/tutorials/portenta-h7/por-ard-lvgl2/content.md:357]:
```suggestion
### 4. Create the Mouse and Keyboard RPC senders
```

Comment (marqdevx) [reply content/tutorials/portenta-h7/por-ard-lvgl2/content.md:357]:
```suggestion
### 4. Create the Mouse and Keyboard RPC senders
```

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md (hunk 3 above)
Comment (marqdevx) [reply content/tutorials/portenta-h7/por-ard-lvgl2/content.md:371]: That wrapper will be in the core or inside the Tutorial's sketches library wrapper

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md (hunk 2 above)
Comment (marqdevx) [reply content/tutorials/portenta-h7/por-ard-lvgl2/content.md:373]:
```suggestion
**Include needed libraries**
```

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md (hunk 4 above)
Comment (marqdevx) [reply content/tutorials/portenta-h7/por-ard-lvgl2/content.md:387]:
```suggestion
**Initialization**
```

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md (hunk 5 above)
Comment (marqdevx) [reply content/tutorials/portenta-h7/por-ard-lvgl2/content.md:397]:
```suggestion
**Mouse handler**
```

File: content/tutorials/portenta-h7/por-ard-lvgl2/content.md (hunk 6 above)
Comment (marqdevx) [reply content/tutorials/portenta-h7/por-ard-lvgl2/content.md:415]:
```suggestion
**Keyboard Handler**
```

**Pull request merged**